addons/mail/static/src/models/thread/thread.js

```diff
@@ -453,17 +453,24 @@ function factory(dependencies) {
         /**
          * Mark the specified conversation as read/seen.
          */
-        async markAsSeen() {
-            if (this.message_unread_counter === 0) {
+        async markAsSeen(message = undefined) {
+            if (!message) {
+                message = this.lastMessage;
+            }
+            if (message.id <= this.pendingSeenMessageId) {
                 return;
             }
+            this.update({ pendingSeenMessageId: message.id });
             if (this.model === 'mail.channel') {
                 const seen_message_id = await this.async(() => this.env.services.rpc({
                     model: 'mail.channel',
                     method: 'channel_seen',
-                    args: [[this.id]]
+                    args: [[this.id]],
+                    kwargs: {
+                        last_message_id: message.id,
+                    },
                 }, { shadow: true }));
-                this.update({ seen_message_id });
+                this.update({ pendingSeenMessageId: seen_message_id, seen_message_id });
```

Comment:
Should set pending to undefined when it is not pending anymore.

Also need to take care that when 2 RPC are done successively, the result from one should not override the result from the other depending on their execution order.

Also if the seen info is received on the bus, the result of the RPC could be ignored all together.

Reply:
> Also if the seen info is received on the bus, the result of the RPC could be ignored all together.

I don't know for the bus

Reply:
Finally I had to depend on bus instead of RPC result as it proved to be more robust